internal/service/device.go

```diff
@@ -292,12 +294,46 @@ func (h *ServiceHandler) PatchDeviceStatus(ctx context.Context, name string, pat
 	return result, StoreErrorToApiStatus(err, false, api.DeviceKind, &name)
 }
 
+func (h *ServiceHandler) HealthcheckDevice(ctx context.Context, name string) api.Status {
+	if err := healthchecker.HealthChecks.Instance().Add(ctx, getOrgIdFromContext(ctx), name); err != nil {
+		h.log.WithError(err).Errorf("failed to add healthcheck to device %s", name)
+		return api.StatusInternalServerError(fmt.Sprintf("failed to add healthcheck to device %s: %v", name, err))
+	}
+	return api.StatusNoContent()
+}
+
 func (h *ServiceHandler) GetRenderedDevice(ctx context.Context, name string, params api.GetRenderedDeviceParams) (*api.Device, api.Status) {
+	var (
+		isNew             bool
+		kvRenderedVersion string
+		err               error
+	)
+
 	orgId := getOrgIdFromContext(ctx)
 
-	result, err := h.store.Device().GetRendered(ctx, orgId, name, params.KnownRenderedVersion, h.agentEndpoint)
-	if err == nil && result == nil {
-		return nil, api.StatusNoContent()
+	if params.KnownRenderedVersion != nil {
+		isNew, kvRenderedVersion, err = rendered_version.Bus.Instance().WaitForNewVersion(ctx, orgId, name, *params.KnownRenderedVersion)
```

Comment:
Why do we have the handler block for 2 minutes waiting for a new version instead of just immediately returning NoContent if there isn't a new version? This feels like an attempt at pushing updates to devices, but hoping that they're attempting a poll when the update occurs. I'm unsure how often we expect real devices to be updating, but I would guess that it's unexpected that every `GetRenderedDevice` results in a new version.

We probably also need to update the agent's poller to remove that log for slow api calls.

Reply:
https://github.com/flightctl/flightctl/blob/main/internal/agent/device/publisher/publisher.go#L112-L121

Reply:
> For steady state scenario we do not expect the devices to be updated often.

Maybe to help us make the decision we can see the actual vs perceived performance gains? Do we have that data? Will having a larger number of concurrent requests open cause any performance issues at scale? Will future load balancers or proxies cause interference at all?

For example I am curious if we could start at say 30-45s. It would be nice to see the metrics at 100k. Starting with 30s then move up to 2m.

Reply:
In general I think we should start lower and increase as required. WDYT?

Reply:
The frequency of the API calls has direct affect on performance.  When measured, 30s interval lead to 90% api-server CPU, while 2 minutes lead to 30% api-server CPU.
Usually we would decrease the interval to gain responsiveness and increase to gain scalability and performance.  With the current implementation we have 2 minutes timeout but with immediate responsiveness.  So why decrease it?